src/main/java/br/codenation/hospital/manager/service/HospitalService.java

```diff
@@ -29,14 +43,80 @@ public Hospital loadHospital(String hospitalId) {
             () -> new ResourceNotFoundException(String.format(HOSPITAL_NOT_FOUND, hospitalId)));
   }
 
-  public Patient save(Patient patient) {
-    return patientRepository.save(patient);
+  public List<Hospital> loadAllHospitals() {
+    return hospitalRepository.findAll();
   }
 
-  public Patient loadPatient(String patientId) {
-    return patientRepository
-        .findById(patientId)
-        .orElseThrow(
-            () -> new ResourceNotFoundException(String.format(PATIENT_NOT_FOUND, patientId)));
+  public Patient checkinPatient(Patient patient, String hospitalId) {
+    Hospital hospital = loadHospital(hospitalId);
+
+    if(!hospital.decAvailableBeds()) throw new HospitalException(String.format(NO_AVAILABLE_BEDS,hospitalId));
+
+    patient.setHospitalCheckIn(LocalDate.now());
+
+    patientService.update(patient);
+
+    if(hospital.addNewPatient(patient) == null) throw new HospitalException(PATIENT_ALREADY_CHECKED_IN);
+
+    save(hospital);
+
+    return patient;
+  }
+
+  public Patient checkoutPatient(String patientId, String hospitalId) {
+    Hospital hospital = loadHospital(hospitalId);
+
+    if(!hospital.incAvailableBeds()) throw new HospitalException(String.format(INVALID_NUMBER_OF_BEDS));
+
+    Patient patient = hospital.removePatient(patientId);
+    if(patient == null) throw new HospitalException(String.format(PATIENT_IS_NOT_CHECKED_IN,patientId));
+
+    patient.setHospitalCheckIn(null);
+    patientService.update(patient);
+    save(hospital);
+
+    return patient;
+  }
+
+  public Patient findPatient(String patientId,String hospitalId){
+    Hospital hospital = loadHospital(hospitalId);
+
+    if(hospital.patientsMapIsNullOrEmpty()) throw new ResourceNotFoundException(String.format(PATIENT_NOT_FOUND, patientId));
+
+    Patient patient = hospital.getPatients().get(patientId);
+
+    if(patient != null) return patient;
+    else throw new ResourceNotFoundException(String.format(PATIENT_NOT_FOUND, patientId));
   }
+
+  public List<Patient> findPatients(String hospitalId){
+    Hospital hospital = loadHospital(hospitalId);
+
+    if(hospital.patientsMapIsNullOrEmpty()) throw new ResourceNotFoundException(String.format(NO_PATIENTS_HAS_BEEN_FOUND));
+
+    return new ArrayList<>(hospital.getPatients().values());
```

Comment:
This is not necessary. We should just return the values.

Reply:
This is how you use it. 
```java
hospitalService.findPatients(hospitalId).stream().
                 .map(Patient::getFullName)
                 .collect(Collectors.toList()));
```

There's no indication it should be a list. It must be streamable, which a collection is.